server/src/main/scala/org/http4s/server/middleware/ResponseTiming.scala

```diff
@@ -0,0 +1,38 @@
+package org.http4s
+package server
+package middleware
+
+import cats.data.Kleisli
+import cats.effect._
+import cats.implicits._
+import org.http4s.util.CaseInsensitiveString
+
+import scala.concurrent.duration._
+
+object ResponseTiming {
+
+  /**
+    * Simple middleware for adding a custom header with timing information to a response.
+    *
+    * This middleware captures the time from when the request headers are parsed and supplied
+    * to the wrapped service to when the response is started. Metrics middleware, like this one,
+    * work best as the outer layer to ensure work done by other middleware is also included.
+    *
+    * @param http [[HttpApp]] to transform
+    * @param timeUnit the units of measure for this timing
+    * @param headerName the name to use for the header containing the timing info
+    */
+  def apply[F[_]: Sync](
+      http: HttpApp[F],
+      timeUnit: TimeUnit = MILLISECONDS,
+      headerName: CaseInsensitiveString = CaseInsensitiveString("X-Response-Time"))(
+      implicit clock: Clock[F]): HttpApp[F] =
+    Kleisli { req =>
+      for {
+        before <- clock.monotonic(timeUnit)
+        resp <- http(req)
+        after <- clock.monotonic(timeUnit)
+        header = Header(headerName.value, s"${after - before}")
+      } yield resp.copy(headers = resp.headers.put(header))
```

Comment:
this could be
```suggestion
      } yield resp.withHeaders(resp.headers.put(header))
```

Reply:
I don't think there is a `withHeaders` available. I can do something like this:
```scala
} yield resp.putHeaders(resp.headers.put(header).toList:_*)
```
Do you think that's better?

Reply:
Seems like an oversight to me.

Reply:
Looks like `putHeaders` was enough.